dftimewolf/lib/collectors/stackdriver.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+"""Reads logs from a GCP cloud project."""
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import os
+import json
+import tempfile
+
+from google.auth import exceptions as google_auth_exceptions
+from google.cloud import logging
+from googleapiclient.errors import HttpError
+from oauth2client.client import AccessTokenRefreshError
+from oauth2client.client import ApplicationDefaultCredentialsError
+
+from dftimewolf.lib import module
+from dftimewolf.lib.containers import StackdriverLogs
+# Need to register with in the protobuf registry.
+from dftimewolf.lib.collectors import audit_log_pb2 as _
+
+
+# Monkey patching the ProtobufEntry because of various issues, notably
+# https://github.com/googleapis/google-cloud-python/issues/7918
+def custom_to_api_repr(self):
+  """API repr (JSON format) for entry."""
+  info = super(logging.entries.ProtobufEntry, self).to_api_repr()
+  info['protoPayload'] = self.payload
+  return info
+
+
+logging.entries.ProtobufEntry.to_api_repr = custom_to_api_repr
+
+
+class StackdriverLogsCollector(module.BaseModule):
+  """Collector for Stackdriver logs."""
+
+  def cleanup(self):
+    pass
+
+  def __init__(self, state):
+    """Initializes a Stackdriver logs collector."""
+    super(StackdriverLogsCollector, self).__init__(state)
+    self._project_name = None
+    self._filter_expression = None
+    self._output_path = tempfile.mkdtemp()
+
+  # pylint: disable=arguments-differ
+  def setup(self, project_name, filter_expression):
+    """Sets up a a Stackdriver logs collector.
+
+    Args:
+      project_name (str): name of the project to fetch logs from.
+      filter_expression (str): Stackdriver advanced logs filter expression.
+    """
+    self._project_name = project_name
```

Comment:
How easy is it with the currently imported libs to check for access to the project here?